### PR TITLE
feat: convert net.atos.zac.app.signaleringen to Kotlin

### PR DESCRIPTION
**Pull request by hsiliev**
We want to use co-routines for async processing, so Kotlin is needed for this

Solves PZ-1937

### CODE REVIEW

File: src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    @Path("/zaken/{type}")
+    fun listZakenSignaleringen(
+        @PathParam("type") signaleringsType: SignaleringType.Type
+    ): List<RESTZaakOverzicht> {
+        val parameters = SignaleringZoekParameters(loggedInUserInstance.get())
+            .types(signaleringsType)
+            .subjecttype(SignaleringSubject.ZAAK)
+        return signaleringenService.listSignaleringen(parameters).stream()
+            .map { signalering: Signalering -> zrcClientService.readZaak(UUID.fromString(signalering.subject)) }
+            .map { zaak: Zaak? -> restZaakOverzichtConverter.convert(zaak!!) }
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt:79]: I think this code needs to be tweaked a bit. `zaak` is treated as nullable here but if you look at the actual zrcClientService you will see that it never returns null (if will throw an exception if it fails to find the zaak). So this line should be: 
```suggestion
            .map { restZaakOverzichtConverter.convert(it) }
```

>@return {@link Zaak}. Never NULL!

File: src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    @GET
+    @Path("/zaken/{type}")
+    fun listZakenSignaleringen(
+        @PathParam("type") signaleringsType: SignaleringType.Type
+    ): List<RESTZaakOverzicht> {
+        val parameters = SignaleringZoekParameters(loggedInUserInstance.get())
+            .types(signaleringsType)
+            .subjecttype(SignaleringSubject.ZAAK)
+        return signaleringenService.listSignaleringen(parameters).stream()
+            .map { signalering: Signalering -> zrcClientService.readZaak(UUID.fromString(signalering.subject)) }
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt:78]: Ok for this PR but we can make code like a this a bit more concise.
```suggestion
            .map { zrcClientService.readZaak(UUID.fromString(it.subject)) }
```

**Review by edgarvonk — COMMENTED** [review]
nice work!

**Comment by edgarvonk** [comment]
Before merging this PR we should switch off the default 'squash commits' setting in GitHub.

File: src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt (hunk 2 above)
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt]: Thanks @edgarvonk ! Fixed

**Review by edgarvonk — APPROVED** [review]
Looks good!

**Review by bas-info-nl — APPROVED** [review]
Some minor comments.

File: src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        val parameters = SignaleringZoekParameters(loggedInUserInstance.get())
+        return signaleringenService.latestSignalering(parameters)
+    }
+
+    @GET
+    @Path("/zaken/{type}")
+    fun listZakenSignaleringen(
+        @PathParam("type") signaleringsType: SignaleringType.Type
+    ): List<RESTZaakOverzicht> {
+        val parameters = SignaleringZoekParameters(loggedInUserInstance.get())
```
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt:69]: Extension functions are great in Kotlin indeed! I keep forgetting about them. Funny, in that 'Java to Kotlin' book the authors also mention how fond they are of extension functions. :-)

File: src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+import net.atos.zac.signalering.model.SignaleringZoekParameters
+import nl.lifely.zac.util.NoArgConstructor
+import java.time.ZonedDateTime
+import java.util.UUID
+
+@Path("signaleringen")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@Singleton
+@NoArgConstructor
```
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/app/signaleringen/SignaleringenRestService.kt:43]: @NoArgConstructor is always needed yes. This has nothing to do with yes or no constructor injection.

**Pull request merged**